central/cluster/datastore/datastore.go

```diff
@@ -128,21 +131,18 @@ func New(
 		notifier:                notifier,
 		clusterRanker:           clusterRanker,
 		networkBaselineMgr:      networkBaselineMgr,
-
-		idToNameCache: simplecache.New(),
-		nameToIDCache: simplecache.New(),
+		clusterCVEDataStore:     clusterCVEs,
+		idToNameCache:           simplecache.New(),
+		nameToIDCache:           simplecache.New(),
 	}
 
 	if features.PostgresDatastore.Enabled() {
 		ds.searcher = search.NewV2(clusterStorage, indexer, clusterRanker)
 	} else {
 		ds.searcher = search.New(clusterStorage, indexer, graphProvider, clusterRanker)
 	}
-	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
-		sac.AllowFixedScopes(
-			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
-			sac.ResourceScopeKeys(resources.Cluster)))
-	if err := ds.buildIndex(ctx); err != nil {
+
+	if err := ds.buildIndex(sac.WithAllAccess(context.Background())); err != nil {
```

Comment:
why the change here?

Reply:
The internal instantiations are completely controlled by us. For instances like building an index on a startup, we really do not benefit from such instantiations. I remember Viswa, working on an issue in the network graph, and the issue was that deployment perm was not included while building the graph.